File: 백준/Silver/5567. 결혼식/결혼식.py
from collections import deque

n=int(input())
m=int(input())
graph=[[] for _ in range(n+1)]
visited=[0]*(n+1)
count=0

for _ in range(m):
    a,b=map(int,input().split())
    graph[a].append(b)
    graph[b].append(a)

def bfs(start):
    q=deque([start])
    visited[start]=1
    while q:
        v=q.popleft()
        for i in graph[v]:
            if not visited[i]:
                q.append(i)
                visited[i]=visited[v]+1

bfs(1)
for i in range(2,n+1):
    if 0<visited[i]<=3:
        count+=1
print(count)

File: 백준/Silver/5567. 결혼식/결혼식.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main{
	static Queue<Integer> q = new LinkedList<>();
	static int graph[][];
	static int visited[];
	static int n,m;
	static int result;
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		n=Integer.parseInt(br.readLine());
		m=Integer.parseInt(br.readLine());
		graph=new int[n+1][n+1];
		visited=new int[n+1];
		for(int i=0;i<m;i++) {
			StringTokenizer st = new StringTokenizer(br.readLine()," ");
			int a=Integer.parseInt(st.nextToken());
			int b=Integer.parseInt(st.nextToken());
			graph[a][b]=graph[b][a]=1;
		}
		bfs(1);
		for(int i=2;i<=n;i++) {
			if(visited[i]<=3 & visited[i]!=0)
				result+=1;
		}
		System.out.println(result);
	}
	public static void bfs(int start) {
		q.add(start);
		visited[start]=1;
		while(!q.isEmpty()) {
			int v = q.poll();
			for(int i=0;i<graph[v].length;i++) {
				if(graph[v][i]==1 & visited[i]==0) {
					q.add(i);
					visited[i]=visited[v]+1;
				}
			}
		}
	}
}